core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java

```diff
@@ -207,25 +212,30 @@ public void init() {
 
     @Override
     public LineTcpConnectionContext of(int fd, @NotNull IODispatcher<LineTcpConnectionContext> dispatcher) {
-        super.of(fd, dispatcher);
-        if (recvBufStart == 0) {
-            recvBufStart = Unsafe.malloc(configuration.getNetMsgBufferSize(), MemoryTag.NATIVE_ILP_RSS);
-            recvBufEnd = recvBufStart + configuration.getNetMsgBufferSize();
-            recvBufPos = recvBufStart;
-            resetParser();
-        }
-        authenticator.init(socket, recvBufStart, recvBufEnd, 0, 0);
-        if (authenticator.isAuthenticated() && securityContext == DenyAllSecurityContext.INSTANCE) {
-            // when security context has not been set by anything else (subclass) we assume
-            // this is an authenticated, anonymous user
-            securityContext = configuration.getFactoryProvider().getSecurityContextFactory().getInstance(
-                    null,
-                    SecurityContext.AUTH_TYPE_NONE,
-                    SecurityContextFactory.ILP
-            );
-            securityContext.authorizeLineTcp();
+        try {
```

Comment:
these contexts are pooled and reused. I can imagine a temporary state of the server when new context cannot be created due to memory pressure. When this happens we close the `authenticator` and potentially leak the object. By that i mean that IODispatchers do not handle Throwable. They handle CairoException. While this code may work as it, it is not a clean workflow.


Better workflow would be to:
- not close the authenticator
- handle Throwable in the IODispatcher
- return object to pool for reuse as normal

Reply:
Attempt of addressing it:

https://github.com/questdb/questdb/pull/4480/commits/3d49bc50e4f551b709e7aebcb57aa0fc0dea8536

Reply:
yep, this works